Revit_Core_Engine/Query/DrawingArea.cs

```diff
@@ -0,0 +1,110 @@
+                compositeGeom.Elements.Add(bhomLine);
+            }
+
+            BH.oM.Geometry.Point centrePoint = compositeGeom.Bounds().Centre();
+
+            var upLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint.Translate(Vector.YAxis * 10));
+            var rightLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint.Translate(Vector.XAxis * 10));
+            var downLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint.Translate(-Vector.YAxis * 10));
+            var leftLine = BH.Engine.Geometry.Create.Line(centrePoint, centrePoint.Translate(-Vector.XAxis * 10));
+
+            List<BH.oM.Geometry.Point> upPoints = new List<BH.oM.Geometry.Point>();
+            List<BH.oM.Geometry.Point> rightPoints = new List<BH.oM.Geometry.Point>();
+            List<BH.oM.Geometry.Point> downPoints = new List<BH.oM.Geometry.Point>();
+            List<BH.oM.Geometry.Point> leftPoints = new List<BH.oM.Geometry.Point>();
+
+            foreach (BH.oM.Geometry.Line bhomLine in compositeGeom.Elements)
+            {
+                var upPoint = upLine.LineIntersection(bhomLine);
+                if (upPoint != null)
+                    upPoints.Add(upPoint);
+
+                var rightPoint = rightLine.LineIntersection(bhomLine);
+                if (rightPoint != null)
+                    rightPoints.Add(rightPoint);
+
+                var downPoint = downLine.LineIntersection(bhomLine);
+                if (downPoint != null)
+                    downPoints.Add(downPoint);
+
+                var leftPoint = leftLine.LineIntersection(bhomLine);
+                if (leftPoint != null)
+                    leftPoints.Add(leftPoint);
+            }
+
+            var areaUpPoint = upPoints.OrderBy(x => x.Distance(centrePoint)).FirstOrDefault();
+            var areaRightPoint = rightPoints.OrderBy(x => x.Distance(centrePoint)).FirstOrDefault();
+            var areaDownPoint = downPoints.OrderBy(x => x.Distance(centrePoint)).FirstOrDefault();
+            var areaLeftPoint = leftPoints.OrderBy(x => x.Distance(centrePoint)).FirstOrDefault();
+
+            List<BH.oM.Geometry.Point> areaPoints = new List<BH.oM.Geometry.Point> { areaUpPoint, areaRightPoint, areaDownPoint, areaLeftPoint };
+
+            var minX = areaPoints.Select(x => x.X).OrderBy(x => x).FirstOrDefault();
+            var minY = areaPoints.Select(x => x.Y).OrderBy(x => x).FirstOrDefault();
+            var maxX = areaPoints.Select(x => x.X).OrderByDescending(x => x).FirstOrDefault();
+            var maxY = areaPoints.Select(x => x.Y).OrderByDescending(x => x).FirstOrDefault();
+
+            var minPoint = BH.Engine.Geometry.Create.Point(minX, minY);
+            var maxPoint = BH.Engine.Geometry.Create.Point(maxX, maxY);
+
+            Outline drawingAreaOutline = new Outline(minPoint.ToRevit(), maxPoint.ToRevit());
+
+            return drawingAreaOutline;
```

Comment:
Could be cleaner to do:

```suggestion
            BoundingBox areaBox = areaPoints.Bounds();
            return new Outline(areaBox.Min.ToRevit(), areaBox.Max.ToRevit());
```